scripts/qa_crowbarsetup.sh

```diff
@@ -5074,27 +5206,77 @@ function oncontroller_testpostupgrade
     fi
 
     for fip in $fips; do
-        scp cirros@$fip:/var/log/ping_neighbour.out ping_neighbour.$fip.out
-        max=$(sed -n 's/^.* not available for: //p' ping_neighbour.$fip.out | sort -n | tail -n 1)
+        scp cirros@$fip:/var/log/ping_neighbour.out ping_neighbour.$fip.out.${timestamp}
+        max=$(sed -n 's/^.* not available for: //p' ping_neighbour.$fip.out.${timestamp} | sort -n | tail -n 1)
         echo "Maximum outage while pinging other VM from $fip: $max seconds"
 
-        scp cirros@$fip:/var/log/ping_outside.out ping_outside.$fip.out
-        max=$(sed -n 's/^.* not available for: //p' ping_outside.$fip.out | sort -n | tail -n 1)
+        scp cirros@$fip:/var/log/ping_outside.out ping_outside.$fip.out.${timestamp}
+        max=$(sed -n 's/^.* not available for: //p' ping_outside.$fip.out .${timestamp}| sort -n | tail -n 1)
         echo "Maximum outage while pinging outside IP from $fip: $max seconds"
 
-        scp cirros@$fip:/mnt/cinder_test.out cinder_test.$fip.out
+        scp cirros@$fip:/mnt/cinder_test.out cinder_test.$fip.out.${timestamp}
         res=$(awk '$1!=p+1{print $1-p}{p=$1}' cinder_test.$fip.out | tail -n +2 | sort | tail -n 1)
         if [ -z "$res" ]; then
             echo "No cinder volume outage when writing from $fip"
         else
             echo "Maximum cinder volume outage when writing from $fip: $res seconds"
         fi
     done
+}
 
+function oncontroller_testpostupgrade_simple_delete_stack
+{
     openstack stack delete --yes upgrade_test
     wait_for 15 20 "! openstack stack show upgrade_test" \
              "heat stack for upgrade tests to be deleted"
-    echo "test post-upgrade successful."
+    echo "test post-upgrade stack deletion successful."
+}
+
+function oncontroller_testpostupgrade_extended_data
+{
+    local stack_names=$(echo $extended_upgrade_templates | sed -e s/2-instances-upgrade-//g -e 's/\.yaml//g')
+    local fips_file=$(mktemp)
+    local timestamp=$(date '+%s')
+
+    for stack in $stack_names
+        do
+        get_stack_fips $stack $fips_file
+        done
+    local fips=$(cat $fips_file)
+    rm $fips_file
+
+    for fip in $fips; do
+        # ensure we don't have host keys from a previous run in known_hosts
```

Comment:
What possible previous run do you mean?

Reply:
If desired, this code can be re-run (e.g. after changing the Cinder configuration to test with different volume backends without redeploying the whole cloud)